test/onyx/integration/s3_test.clj

```diff
@@ -0,0 +1,93 @@
+(ns onyx.integration.s3-test
+  (:require [amazonica.aws.s3 :as s3c]
+            [clojure.test :refer :all]
+            [schema.test :as st]
+            [aero.core :refer [read-config]]
+            [clojure.core.async :refer [<!! >!! alts!! timeout chan pipe]]
+            [clojure.core.async.lab :refer [spool]]           
+            [io.aviso.exception :as pretty]
+            [onyx
+             [api]
+             [job :refer [add-task]]]
+            [onyx.plugin
+             [s3 :as s3p]
+             [core-async :refer [get-core-async-channels]]]
+            [onyx.tasks
+             [core-async :as core-async]
+             [s3 :as s3]])
+  (:import [com.amazonaws.services.s3.model S3Object]
+           [java.io ByteArrayInputStream]))
+
+(defn uuid
+  []
+  (str (java.util.UUID/randomUUID)))
+
+(defn unique-s3-file-name
+  []
+  (str "onyx-s3-plugin/" (uuid)))
+
+(def peer-config
+  {:zookeeper/address "localhost:2181"
+   :onyx/tenancy-id "1"
+   :onyx.peer/job-scheduler :onyx.job-scheduler/balanced
+   :onyx.messaging/impl :aeron
+   :onyx.messaging/peer-port 40200
+   :onyx.messaging/bind-addr "localhost"})
+
+(def creds {:access-key "ADD_YOUR_OWN" ;ditto for below
+            :secret-key "ADD_YOUR_OWN"
+            :endpoint "eu-west-1"})
```

Comment:
I think it's a bad idea to invite people to add their AWS creds. Can we either use environ or just let amazonica pick them up and advise people via the README how to get their creds in?